packages/babel-make-styles/src/utils/evaluatePathsInVM.ts

```diff
@@ -63,7 +63,19 @@ const expressionWrapperTpl = template.statement(`
   };
 `);
 
-const expressionTpl = template.expression(`%%wrapName%%(() => %%expression%%)`);
+/**
+ * Functions, call & member expressions should be wrapped with IIFE to ensure that "theme" object will be passed
+ * without collisions.
+ *
+ * @example
+ * {
+ *   label: foo(), // call expression
+ *   header: typography.header, // could be an object or a function
+ * }
+ */
+const expressionTpl = template.expression(
+  `%%wrapName%%(() => typeof %%expression%% === 'function' ? %%expression%%(%%themeVariableName%%) : %%expression%%)`,
```

Comment:
do you mind adding an example of the expression once it has been formatted ?

Reply:
Added in cf52f5f, also added a test 👍